apiserver/facades/client/charms/charmhubrepo.go

```diff
@@ -120,19 +140,24 @@ func (c *chRepo) ResolveWithPreferredChannel(curl *charm.URL, origin params.Char
 	// Ensure we send the updated curl back, with all the correct segments.
 	revision := entity.Revision
 	resCurl := curl.
-		WithSeries(origin.Series).
-		WithArchitecture(origin.Architecture).
+		WithSeries(input.Platform.Series).
+		WithArchitecture(input.Platform.Architecture).
 		WithRevision(revision)
 
 	// Create a resolved origin.  Keep the original values for ID and Hash, if any
 	// were passed in.  ResolveWithPreferredChannel is called for both charms to be
 	// deployed, and charms which are being upgraded.  Only charms being upgraded
 	// will have an ID and Hash.  Those values should only ever be updated in
```

Comment:
Apparently I left a dangling sentence in the comments, could you please add:
     // chRepro FindDownloadURL.
for me?